[docs-infra] Fix intersection observer for demo code blocks

dav-is opened the pull request:
Frames within demo code blocks are not being properly initialized in the intersection observer.

It seems we were adding each child of the `<pre>`, meaning the single `<code>` element, instead of each span frame within `<code>`:

```html
<pre>
  <code> <-- target before
    <span class="frame" data-frame="0">...</span> <-- target after PR
    <span class="frame" data-frame="1">...</span> <-- target after PR
  </code>
</pre>
```

<img width="1886" height="1310" alt="Screenshot 2025-12-11 at 15 04 58" src="https://github.com/user-attachments/assets/fc11635d-0806-4031-9532-cfca75c0b47d" />

Janpot reviewed (approved):
Perhaps
```tsx
root.getElementsByClassName('frame')
```
is more robust? Probably much slower tough.

dav-is commented:
> Probably much slower tough

Right, it would have check the entire highlighted text, when we know the structure of the html already.

Added this as a sanity check that shouldn't be too expensive:
```ts
 if (!element.hasAttribute('data-frame')) {
    console.warn('Expected frame element in useCode <Pre>', element);
    return;
}
```

Also, updated prettier to keep it in sync with `code-infra`: https://github.com/mui/base-ui/pull/3499#discussion_r2610725331
